scripts/loki_transform.py

```diff
@@ -268,10 +270,19 @@ def transform_subroutine(self, routine, **kwargs):
         scheduler.process(transformation=HoistTemporaryArraysAnalysis(dim_vars=(vertical.size,)))
         scheduler.process(transformation=HoistTemporaryArraysDeviceAllocatableTransformation())
 
-    # Housekeeping: Inject our re-named kernel and auto-wrapped it in a module
-    scheduler.process( ModuleWrapTransformation(module_suffix='_MOD') )
-    mode = mode.replace('-', '_')  # Sanitize mode string
-    scheduler.process( DependencyTransformation(suffix=f'_{mode.upper()}', module_suffix='_MOD') )
+    mode = mode.replace('-', '_')  # Sanitize mode string    
+    if mode in ["c"]:
+        f2c_transformation = FortranCTransformation(path=build)
+        scheduler.process(f2c_transformation)
+        for h in definitions:
+            f2c_transformation.apply(h, role='header')
```

Comment:
Ah, here it is applied directly to a file, but the F2C Transformation is not marked as recursing to internal scopes - instead it does this internally. I think we should take this opportunity to make the Transformation compliant with the manifest parameterisation. I'll explain in the overall review comment what I mean.

Secondly, I'm not too thrilled by the fact that we provide files as headers (which, to me, implies a "read-only for enrichment" meaning) and then apply transformations to them. Since the purpose here is to deal with derived types, which we cannot capture in the dependency graph in the current Scheduler implementation, we may have to live with this for now, but that will be different with #213. To avoid that others copy this pattern, would you mind including a comment here that explains that this is temporary?
Longer term, this transformation step to generate accessor methods for derived types should become a separate Transformation that is applied to `ModuleItem` only.